examples/apps/meshdemo/kubernetes/app/kustomize/base/colors.yaml

```diff
@@ -0,0 +1,255 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: colors
+  labels:
+    name: colors
+  namespace: meshdemo
+spec:
+  ports:
+  - port: 8080
+  selector:
+    app: colors
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: colors-blue
+  labels:
+    name: colors-blue
+  namespace: meshdemo
+spec:
+  ports:
+  - port: 8080
+  selector:
+    name: colors-blue
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: colors-blue
+  namespace: meshdemo
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: colors-blue
+        app: colors
+    spec:
+      serviceAccountName: xray-access
+      containers:
+      - name: colors-blue
+        image: 072792469044.dkr.ecr.eu-west-1.amazonaws.com/meshdemo/colors:1.0.0
+        ports:
+        - containerPort: 8080
+        args: ["--color", "blue"]
+      - name: xray-daemon
+        image: 072792469044.dkr.ecr.eu-west-1.amazonaws.com/temporary/aws-xray-daemon:0.5
+        securityContext:
+          runAsUser: 1337
+        ports:
+        - containerPort: 2000
+          name: xray
+          protocol: UDP
+        env:
+        - name: AWS_REGION
+          value: eu-west-1
+        resources:
+          requests:
+            cpu: 100m
+            memory: 128Mi
+      - name: envoy
+        env:
+        - name: APPMESH_VIRTUAL_NODE_NAME
+          value: mesh/meshdemo/virtualNode/colors-blue-meshdemo
+        - name: ENVOY_LOG_LEVEL
+          value: info
+        - name: AWS_REGION
+          value: eu-west-1
+        - name: ENABLE_ENVOY_XRAY_TRACING
+          value: "1"
+        image: 840364872350.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-envoy:v1.12.1.0-prod
```

Comment:
We make an effort to not hardcode any envoy images I'm in this repo. Also v1.12.1.0 doesn't support IRSA.

Reply:
Agreed, I actually posted in the internal channel about this.  I had no insight into what the envoy container does and does not support specifically in regards to IRSA.  this is also why eksctl had to set the --appmesh flag.